File: tests/fixtures/source_files/calculator.py
"""Simple Python calculator for testing"""

class Calculator:
    def __init__(self):
        self.result = 0.0

    def add(self, value):
        """Add value to current result"""
        self.result += value
        return self.result

    def subtract(self, value):
        """Subtract value from current result"""
        self.result -= value
        return self.result

    def multiply(self, value):
        """Multiply current result by value"""
        self.result *= value
        return self.result

    def divide(self, value):
        """Divide current result by value"""
        if value == 0:
            raise ValueError("Division by zero")
        self.result /= value
        return self.result

    def get_result(self):
        """Get current result"""
        return self.result

    def clear(self):
        """Reset result to zero"""
        self.result = 0.0

def calculate_area(length, width):
    """Calculate rectangle area"""
    if length <= 0 or width <= 0:
        raise ValueError("Dimensions must be positive")
    return length * width

File: tests/fixtures/source_files/calculator.java
/**
 * Simple Java calculator for testing
 */
public class Calculator {
    private double result;

    public Calculator() {
        this.result = 0.0;
    }

    public double add(double value) {
        this.result += value;
        return this.result;
    }

    public double subtract(double value) {
        this.result -= value;
        return this.result;
    }

    public double multiply(double value) {
        this.result *= value;
        return this.result;
    }

    public double divide(double value) throws ArithmeticException {
        if (value == 0.0) {
            throw new ArithmeticException("Division by zero");
        }
        this.result /= value;
        return this.result;
    }

    public double getResult() {
        return this.result;
    }

    public void clear() {
        this.result = 0.0;
    }

    public static double calculateArea(double length, double width) {
        if (length <= 0 || width <= 0) {
            throw new IllegalArgumentException("Dimensions must be positive");
        }
        return length * width;
    }
}